File: src/main/java/com/procurement/point/repository/ReleaseBudgetRepository.java
package com.procurement.point.repository;

import com.procurement.point.model.entity.ReleaseEntity;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.Date;
import java.util.List;
import java.util.Optional;

@Repository
public interface ReleaseBudgetRepository extends CassandraRepository<ReleaseEntity, String> {

    @Query(value = "select * from notice_budget_compiled_release where cp_id=?0")
    List<ReleaseEntity> getAllByCpId(String cpId);

    @Query(value = "select * from notice_budget_compiled_release where cp_id=?0 and release_date>=?1 ALLOW FILTERING")
    List<ReleaseEntity> getAllByCpIdAndOffset(String cpId, Date offset);

    @Query(value = "select * from notice_budget_release where cp_id=?0 and oc_id=?1")
    List<ReleaseEntity> getAllByCpIdAndOcId(String cpId, String ocId);

    @Query(value = "select * from notice_budget_release where cp_id=?0 and oc_id=?1 and release_date>=?2 ALLOW FILTERING")
    List<ReleaseEntity> getAllByCpIdAndOcIdAndOffset(String cpId, String ocId, Date offset);


}

File: src/main/kotlin/com/procurement/point/repository/ReleaseBudgetRepository.kt
package com.procurement.point.repository

import com.procurement.point.model.entity.ReleaseBudgetEntity
import org.springframework.data.cassandra.repository.CassandraRepository
import org.springframework.data.cassandra.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ReleaseBudgetRepository : CassandraRepository<ReleaseBudgetEntity, String> {

    @Query(value = "select * from notice_budget_compiled_release where cp_id=?0")
    fun getAllCompiledByCpId(cpId: String): List<ReleaseBudgetEntity>

    @Query(value = "select * from notice_budget_compiled_release where cp_id=?0 and release_date>=?1 ALLOW FILTERING")
    fun getAllCompiledByCpIdAndOffset(cpId: String, offset: Date): List<ReleaseBudgetEntity>

    @Query(value = "select * from notice_budget_compiled_release where cp_id=?0 and oc_id=?1 LIMIT 1")
    fun getCompiledByCpIdAndOcid(cpId: String, ocId: String): ReleaseBudgetEntity?

    @Query(value = "select * from notice_budget_release where cp_id=?0 and oc_id=?1")
    fun getAllReleasesByCpIdAndOcId(cpId: String, ocId: String): List<ReleaseBudgetEntity>

    @Query(value = "select * from notice_budget_release where cp_id=?0 and oc_id=?1 and release_date>=?2 ALLOW FILTERING")
    fun getAllReleasesByCpIdAndOcIdAndOffset(cpId: String, ocId: String, offset: Date): List<ReleaseBudgetEntity>


}